### ISSUE TITLE
DB::Exception: pqxx::usage_error: Started new transaction while transaction was still active.. Stack trace:

### ISSUE THREAD

**Issue by trollhe** [ClickHouse/ClickHouse#23058]
When running the insert operation, the following error is reported.

```
Traceback (most recent call last):
  File "rpt_resource_answer_distribution_v1.py", line 867, in <module>
    main(task_name=args.task_name)
  File "rpt_resource_answer_distribution_v1.py", line 844, in main
    rpt_resource_answer_distribution_1min(_execution_date, supplement_number)
  File "rpt_resource_answer_distribution_v1.py", line 330, in rpt_resource_answer_distribution_1min
    raw_aggragation(src_table_name, dst_table_name
  File "rpt_resource_answer_distribution_v1.py", line 199, in raw_aggragation
    result = ch.execSql(sql)
  File "/opt/wavepoint/data-engine/external-libraries/engine_commons/clickhouse/CHUtils.py", line 61, in execSql
    result = client.execute(sql)
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/client.py", line 220, in execute
    rv = self.process_ordinary_query(
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/client.py", line 346, in process_ordinary_query
    return self.receive_result(with_column_types=with_column_types,
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/client.py", line 89, in receive_result
    return result.get_result()
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/result.py", line 50, in get_result
    for packet in self.packet_generator:
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/client.py", line 101, in packet_generator
    packet = self.receive_packet()
  File "/opt/wavepoint/data-engine/external-libraries/clickhouse_driver/client.py", line 118, in receive_packet
    raise packet.exception
clickhouse_driver.errors.ServerException: Code: 1001.
DB::Exception: pqxx::usage_error: Started new transaction while transaction was still active.. Stack trace:

0. pqxx::internal::check_unique_register(void const*, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >, void const*, std::__1::basic_string_view<char, std::__1::char_traits<char> >, std::__1::basic_string_view<char, std::__1::char_traits<char> >) @ 0x138d94ab in /usr/bin/clickhouse
1. pqxx::internal::basic_transaction::basic_transaction(pqxx::connection&, pqxx::zview) @ 0x138d3add in /usr/bin/clickhouse
2. DB::PostgreSQLBlockInputStream::readPrefix() @ 0xe8b8b33 in /usr/bin/clickhouse
3. DB::SourceFromInputStream::generate() @ 0xfaac09e in /usr/bin/clickhouse
4. DB::ISource::tryGenerate() @ 0xf8f9e65 in /usr/bin/clickhouse
5. DB::ISource::work() @ 0xf8f9a5a in /usr/bin/clickhouse
6. DB::SourceFromInputStream::work() @ 0xfaabea4 in /usr/bin/clickhouse
7. ? @ 0xf933e4d in /usr/bin/clickhouse
8. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0xf930a71 in /usr/bin/clickhouse
9. ? @ 0xf935646 in /usr/bin/clickhouse
10. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x864d89f in /usr/bin/clickhouse
11. ? @ 0x8651333 in /usr/bin/clickhouse
12. start_thread @ 0x82de in /usr/lib64/libpthread-2.28.so
13. clone @ 0xfbe83 in /usr/lib64/libc-2.28.so
```

operation env:
```
os: alpine 3.12.x
ch:  clickhouse 21.3.2.5
```

**Comment by trollhe** [comment]
i need multiple task to operation database,how i do?

**Comment by kssenii** [comment]
@trollhe, this was fixed in version 21.4.

**Comment by trollhe** [comment]
> @trollhe, this was fixed in version 21.4. (Fixed in #21839)

thanks
